File: akidocs-core/src/akidocs_core/tokens.py
from dataclasses import dataclass


@dataclass
class Text:
    content: str


@dataclass
class Italic:
    content: str


InlineToken = Text | Italic


@dataclass
class Header:
    level: int
    content: list[InlineToken]


@dataclass
class Paragraph:
    content: list[InlineToken]


Token = Header | Paragraph

File: akidocs-core-rs/src/tokens.rs
pub struct Header {
    pub level: u8,
    pub content: String, // Will become Vec<InlineText> later
}

#[derive(Debug, PartialEq)]
pub struct Paragraph {
    pub content: String, // Will become Vec<InlineText> later
}

#[derive(Debug, PartialEq)]
pub enum Token {
    Paragraph(Paragraph),
    Header(Header),
}
